subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithInstantExecutionRule.groovy

```diff
@@ -32,7 +32,11 @@ class UnsupportedWithInstantExecutionRule implements TestRule {
         if (!GradleContextualExecuter.isInstant() || annotation == null) {
             return base
         }
-        return new SkippingRuleStatement(base)
+        String[] bottomSpecs = annotation.bottomSpecs()
+        if (bottomSpecs.length == 0 || bottomSpecs.any { description.className.endsWith(it) }) {
```

Comment:
❌ Extract a predicate function to keep the `bottomSpecs.length == 0` rule in a single place?